operator/pkg/ciliumendpointbatch/manager.go

```diff
@@ -0,0 +1,422 @@
+				c.cacheCepMap.insert(getCepNameFromCCEP(&cep), srcCeb.GetName())
+			}
+		}
+	}
+}
+
+// If available, getCebFromCache returns CiliumEndpointBatch object.
+func (c *cebManagerFcfs) getCebFromCache(cebName string) (*cilium_v2.CiliumEndpointBatch, error) {
+	if ceb, ok := c.cacheCeb[cebName]; ok {
+		outCeb := new(cilium_v2.CiliumEndpointBatch)
+		ceb.backendMutex.Lock()
+		ceb.ceb.DeepCopyInto(outCeb)
+		ceb.backendMutex.Unlock()
+		return outCeb, nil
+
+	}
+	return nil, fmt.Errorf("Failed to get CEB from local cache for the cebName: %s", cebName)
+}
+
+// findCeb returns the available ceBatch object for a CEP, CEB is chosen based on FCFS.
+// if all CEB's reached max capacity or marked for delete. Allocate a new CEB.
+// if new ceBatch is created, it returns ceBatch object and true
+// else return existing ceBatch and false.
+func (c *cebManagerFcfs) findCeb(cep *cilium_v2.CoreCiliumEndpoint) (*ceBatch, bool) {
+
+	// Get the first available CEB
+	for _, ceb := range c.cacheCeb {
+		// Note: If CEB's is marked for delete or no CEP's present in it don't add
+		// any more new CEP's in it.
+		if len(ceb.ceb.Endpoints) >= c.maxCepsInCeb || len(ceb.ceb.Endpoints) == 0 ||
+			c.cebsToSync.isCebMarkedForDelete(ceb.ceb.GetName()) {
+			continue
+		}
+		return ceb, false
+	}
+
+	// Allocate a new ceBatch and return
+	return c.createCeb(""), true
+}
+
+// InsertCepInCache is used to Insert CEP in local cache, this may result in Creating a New
+// CEB object or Updating an existing CEB object.
+func (c *cebManagerFcfs) InsertCepInCache(cep *cilium_v2.CoreCiliumEndpoint) (string, error) {
+
+	// Check the given cep is already packed in any of the CEB.
+	// if yes, Update a ceb with the given cep object.
+	if cebName, ok := c.cacheCepMap.get(getCepNameFromCCEP(cep)); ok {
+		queueCep(cep, c.cacheCeb[cebName])
+		c.cebsToSync.updateAggregator(cebName, CebUpdate)
+		return cebName, nil
+	}
+
+	cb, isNewCeb := c.findCeb(cep)
+
+	// Cache CEP name with newly allocated CEB.
+	c.cacheCepMap.insert(getCepNameFromCCEP(cep), cb.ceb.GetName())
+
+	// Queue the CEP in CEB
+	queueCep(cep, cb)
+
+	// If it isNewCeb, update Aggregator to create a ceb
+	if isNewCeb {
+		c.cebsToSync.updateAggregator(cb.ceb.GetName(), CebCreate)
+		return cb.ceb.GetName(), nil
+	}
+
+	c.cebsToSync.updateAggregator(cb.ceb.GetName(), CebUpdate)
+	return cb.ceb.GetName(), nil
+}
+
+// RemoveCepFromCache is used to remove CEP from local cache, this may result in
+// Updating an existing CEB object.
+func (c *cebManagerFcfs) RemoveCepFromCache(cep *cilium_v2.CoreCiliumEndpoint) error {
+
+	// Check in local cache, if a given cep is already batched in one of the ceb.
+	// and if exists, delete cep from ceb.
+	cepName := getCepNameFromCCEP(cep)
+	log.WithFields(logrus.Fields{
+		"cep-name": cepName,
+	}).Debug("Remove CEP from local cache")
+	if cebName, ok := c.cacheCepMap.get(cepName); ok {
+		ceb, err := c.cacheCeb[cebName]
```

Comment:
s/err/ok/g

Calling the second returned value `err` is very confusing as it is the exact opposite of error...

Reply:
changed `err` variable name. Reusing `ok` would cause more confusion.